### PR TITLE
Use healthcheck.sh

### LINKED ISSUE #405: Extend MariaDB Container Entrypoint coverage
We pull the mariadb entrypoint script from the canonical image on dockerhub. That entrypoint script also has support for some kind of healthcheck (that we do not test). It would would be great to extend the test coverage to be able to test: https://github.com/SUSE/BCI-dockerfile-generator/pull/904

### PR DESCRIPTION
**Pull request by alexandrevicenzi**
The MariaDB image now provides the healthcheck.sh that can be used to check if the DB is properly initialized. Use this script to ensure containers are ready in the test.

Add tests to check if healthcheck.sh is working as expected.

Fixes #405.

### CODE REVIEW

**Review by dcermak — COMMENTED** [review]
LGTM, but please do add docstrings to the new tests explaining roughly what you are testing there, so that future you or future me can lookup what innodb or galera mean

File: tests/test_mariadb.py [hunk 1]
```diff
@@ -208,3 +195,43 @@ def wait_for_server():
...
+    Test if InnoDB (storage engine) has completed initializing.
+    """
+    conn = auto_container_per_test.connection
+
+    _wait_for_server(conn)
+
+    # the query is similar to: healthcheck.sh --innodb_initialized
+    conn.check_output(
+        f"mariadb -p{MARIADB_ROOT_PASSWORD} -e \"select 1 from information_schema.ENGINES WHERE engine='innodb' AND support in ('YES', 'DEFAULT', 'ENABLED')\""
+    )
```
Comment (dcermak) [inline tests/test_mariadb.py:237]: Why are you adding this test case if it performs the same action as `healthcheck.sh --innodb_initialized`?

File: tests/test_mariadb.py [hunk 2]
```diff
@@ -208,3 +195,43 @@ def wait_for_server():
...
+    _wait_for_server(conn)
+
+    conn.check_output("healthcheck.sh --su-mysql --innodb_initialized")
+
+
+def test_mariadb_healthcheck_galera_cluster_disabled(auto_container_per_test):
+    """
+    Test if Galera Cluster (multi-primary cluster) is disabled (experimental feature).
+
+    See: https://mariadb.com/kb/en/using-healthcheck-sh/#-galera_online
```
Comment (dcermak) [inline tests/test_mariadb.py:217]:
```suggestion
    See: `<https://mariadb.com/kb/en/using-healthcheck-sh/#-galera_online>`_
```

File: tests/test_mariadb.py [hunk 3]
```diff
@@ -208,3 +195,43 @@ def wait_for_server():
     assert rows and len(rows) == 2
     _, num, data = rows[-1].split()
     assert num == "100" and data == "abcdef"
+
+
+def test_mariadb_healthcheck_innodb_initialized(auto_container_per_test):
+    """
+    Test if InnoDB (storage engine) has completed initializing.
+
+    See: https://mariadb.com/kb/en/using-healthcheck-sh/#-innodb_initialized
```
Comment (dcermak) [inline tests/test_mariadb.py:204]: Converts this into a clickable link in the sphinx documentation
```suggestion
    See: `<https://mariadb.com/kb/en/using-healthcheck-sh/#-innodb_initialized>`_
```

File: tests/test_mariadb.py [hunk 4]
```diff
@@ -208,3 +195,43 @@ def wait_for_server():
...
+    conn = auto_container_per_test.connection
+
+    _wait_for_server(conn)
+
+    conn.check_output("healthcheck.sh --su-mysql --innodb_initialized")
+
+
+def test_mariadb_healthcheck_galera_cluster_disabled(auto_container_per_test):
+    """
+    Test if Galera Cluster (multi-primary cluster) is disabled (experimental feature).
```
Comment (dcermak) [inline tests/test_mariadb.py:215]:
```suggestion
    Ensure that Galera Cluster (multi-primary cluster) is disabled (experimental feature) i.e. :command:`healthcheck.sh --su-mysql --galera_online` fails.
```

File: tests/test_mariadb.py (hunk 1 above)
Comment (alexandrevicenzi) [reply tests/test_mariadb.py:237]: The main idea was to check if the healthcheck.sh script ever changes behavior.

Comment (dcermak) [reply tests/test_mariadb.py:237]: I am not terribly enthusiastic about this kind of test to be honest. We'll be testing implementation details which I'd consider a bit of an anti-pattern. I'd add this if we had a regression involving this, but I'm not aware of any. However, I'm open to other points of view.

Comment (alexandrevicenzi) [reply tests/test_mariadb.py:237]: I don't mind removing it. In theory, if the behavior of healthcheck.sh changes, the test are very likely to fail.

File: tests/test_mariadb.py [hunk 5]
```diff
@@ -224,3 +211,30 @@ def wait_for_server():
...
+
+    conn.check_output("healthcheck.sh --su-mysql --innodb_initialized")
+
+
+def test_mariadb_healthcheck_galera_cluster_disabled(auto_container_per_test):
+    """
+    Ensure that Galera Cluster (multi-primary cluster) is disabled (experimental feature),
+    i.e. :command:`healthcheck.sh --su-mysql --galera_online` fails.
+
+    See: <https://mariadb.com/kb/en/using-healthcheck-sh/#-galera_online>
```
Comment (dcermak) [inline tests/test_mariadb.py:234]: Not a proper rst link:
```suggestion
    See: `<https://mariadb.com/kb/en/using-healthcheck-sh/#-galera_online>`_
```

File: tests/test_mariadb.py [hunk 6]
```diff
@@ -224,3 +211,30 @@ def wait_for_server():
     assert rows and len(rows) == 2
     _, num, data = rows[-1].split()
     assert num == "100" and data == "abcdef"
+
+
+def test_mariadb_healthcheck_innodb_initialized(auto_container_per_test):
+    """
+    Test if InnoDB (storage engine) has completed initializing.
+
+    See: <https://mariadb.com/kb/en/using-healthcheck-sh/#-innodb_initialized>
```
Comment (dcermak) [inline tests/test_mariadb.py:220]: Not a proper rst link:
```suggestion
    See: `<https://mariadb.com/kb/en/using-healthcheck-sh/#-innodb_initialized>`_
```

**Review by dcermak — COMMENTED** [review]
LGTM, minor doc nitpick left

**Comment by dcermak** [comment]
@alexandrevicenzi can you please rebase?

**Comment by alexandrevicenzi** [comment]
@dcermak done

**Pull request merged**